timelock-impl/src/main/java/com/palantir/atlasdb/timelock/management/DisabledNamespaces.java

```diff
@@ -90,40 +92,60 @@ public ReenableNamespacesResponse reEnable(ReenableNamespacesRequest request) {
+            Map<Namespace, UUID> lockedNamespaces = KeyedStream.of(namespaces)
+                    .map(Namespace::get)
+                    .map(this::getLockId)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .collectToMap();
+
+            if (!lockedNamespaces.isEmpty()) {
                 log.error(
                         "Failed to disable namespaces, as some were already disabled",
                         SafeArg.of("namespaces", namespaces),
-                        SafeArg.of("alreadyDisabledNamespaces", alreadyDisabled));
-                return DisableNamespacesResponse.unsuccessful(
-                        UnsuccessfulDisableNamespacesResponse.of(alreadyDisabled));
+                        SafeArg.of("lockId", lockId),
+                        SafeArg.of("lockedNamespace", lockedNamespaces));
+                return SingleNodeUpdateResponse.of(false, lockedNamespaces);
             }
 
             Set<String> namespaceNames = namespaces.stream().map(Namespace::get).collect(Collectors.toSet());
             disable(namespaceNames, lockId);
             log.info("Successfully disabled namespaces", SafeArg.of("namespaces", namespaces));
-            return DisableNamespacesResponse.successful(SuccessfulDisableNamespacesResponse.of(lockId));
+            return SingleNodeUpdateResponse.builder().wasSuccessful(true).build();
         }
 
         @Transaction
-        default ReenableNamespacesResponse reEnableAll(Set<Namespace> namespaces, UUID lockId) {
-            Set<Namespace> namespacesWithLockConflict = namespaces.stream()
-                    .filter(ns -> !getLockId(ns.get()).orElse(lockId).equals(lockId))
-                    .collect(Collectors.toSet());
+        default SingleNodeUpdateResponse reEnableAll(Set<Namespace> namespaces, UUID lockId) {
+            Map<Namespace, UUID> namespacesWithLockConflict = KeyedStream.of(namespaces)
+                    .map(Namespace::get)
+                    .map(this::getLockId)
+                    .flatMap(Optional::stream)
+                    .filter(val -> !val.equals(lockId))
+                    .collectToMap();
+
             if (!namespacesWithLockConflict.isEmpty()) {
                 log.error(
-                        "Failed to re-ensable namespaces, as some were disabled with a different lock ID",
+                        "Failed to re-enable namespaces, as some were disabled with a different lock ID",
                         SafeArg.of("namespaces", namespaces),
-                        SafeArg.of("failedNamespaces", namespacesWithLockConflict));
-                return ReenableNamespacesResponse.of(false, namespacesWithLockConflict);
+                        SafeArg.of("expectedLockId", lockId),
+                        SafeArg.of("conflictingNamespaces", namespacesWithLockConflict));
+                return SingleNodeUpdateResponse.of(false, namespacesWithLockConflict);
```

Comment:
This is kind of sketchy -- if we have a race condition, we are guaranteed to be stuck now. I would do best effort and still unlock our locks.

Reply:
changed to unlock non-conflicting namespaces